[Recognizers-Text] Port BaseTimeZoneExtractor methods declaration

Konigstiger opened the pull request:
### Changes made:

We added the **BaseTimeZoneExtractor** class and the following methods declaration:

- extractor_type_name
- extractor_name
- extract
- remove_ambiguous_time_zone
- match_location_times
- match_timezones

Maihav committed:
WIP: Add new BaseTimePeriodParser class with methods

Maihav committed:
WIP add methods declaration

Maihav committed:
WIP: remove unused import & add default value on method

Commit:
Update base_time_zone.py

Changes
- Add stub class BaseTimeZoneExtractor

Commit:
Update base_time_zone.py

Added line

Maihav committed:
WIP: Add missing constants

Maihav committed:
WIP: Add compute_minutes implementation

Maihav committed:
WIP: Update compute_minutes

Maihav committed:
WIP: Add convert_offset_in_mins_to_offset_string implementation

Maihav committed:
WIP: Add convert_offset_in_mins_to_offset_string implementation

Maihav committed:
WIP: Add convert_mins_to_regular_format implementation & missing decorators

Maihav committed:
WIP: Add convert_mins_to_regular_format implementation & missing decorator

Maihav committed:
Finish convert_mins_to_regular_format implementation

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,62 @@
+
+    @property
+    def extractor_name(self) -> str:
+        return Constants.SYS_DATETIME_TIMEZONE
```
_extractor_type_name_ and _extractor_name_  props has the same return value. why? Can you check this?

Konigstiger replied:
Sure. We will be using the extractor_type_name property, and removing the previous extractor_name property. That property is used in the original code, but is better to only use extractor_type_name. Thanks.

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,62 @@
+
+    def match_location_times(self, text: str, tokens: List[Token]) -> List[Token]:
+        ret = []
+        return ret
```
You should add the 'pass' placeholder here. It is used when a statement is required syntactically

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,62 @@
+
+    def match_timezones(self, text: str) -> List[Token]:
+        ret = []
+        return ret
```
Same here

Maihav reviewed (changes requested):
Minor feedback in line. Also I noticed that you missed the config prop declaration and the constructor. Please, check it

Commit:
Update BaseTimeZoneExtractor, Add TimeZoneExtractorConfiguration

Changes
- Applied feedback from previous PR.
- Removed unnecesary property extractor?name.
- Added required class TimeZoneExtractorConfiguration

Maihav committed:
Add normalize_text implementation

Maihav committed:
Apply PRss feedback

Maihav committed:
Add parse method implementation

Commit:
Update BaseTimeZoneExtractor: match_timezones

Changes
- First implementation of method match_timezones
- This is a work in progress.

Commit:
Update base_time_zone.py

Add line

Maihav committed:
WIP: Implement Parse method

Maihav committed:
WIP: Add TimeZoneResolutionResult and update DateTimeResolutionResult implementation

Maihav committed:
Fix autopep

Maihav committed:
Add new line at end of file

Maihav committed:
Fix autopep

Maihav committed:
Fix pep8

Maihav committed:
Implement filter_results

Maihav committed:
Add abstract method

Maihav committed:
Update resource-definitions to generate TimeZone resource

Commit:
WIP: Updates to BaseTimeZoneExtractor, QueryProcessor, and Resource Definitions

Changes
*BaseTimeZoneExtractor*
- extract
- match_location_times
- match_timezones

QueryProcessor
- remove_diacritics

Resource Definitions

Commit:
PEP8

Maihav committed:
Import TimeZoneDefinitions on init file

Maihav committed:
Add newline

Commit:
Import TimeZoneDefinitions on init file

Maihav committed:
Finish to port Parse method

Maihav committed:
Add TimeZone constant

Maihav committed:
Delete semicolon

Maihav committed:
Create TimeZoneExtractorConfiguration class

Maihav committed:
Add missing import and newline

Maihav committed:
Import base_timezone

Maihav committed:
Create EnglishDateExtractorConfiguration class & Add props

Maihav committed:
WIP: Add EnglishDateExtractorConfiguration constructor & initialize props

Maihav committed:
Fix autopep

Maihav committed:
Add remove_diacritics methods on QueryProcessor class

Commit:
Updated BaseTimeZoneExtractor

Changes
- Ported completed version of method match_location_times

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,153 @@
+            for match in direct_utc:
+                result.append(Token(match.start(), match.end() + match.start()))
+
+            matches = self.config.time_zone_matcher.find(text)
```
why are you using find() here? While find() retrieves an integer, finditer() returns an iterator of MatchObject instances. Can you review it?

Commit:
Update BaseTimeZoneExtractor

Changes
- Ported method extract, that references previous ported methods.

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,153 @@
+
+        if len(time_match) != 0 and not is_all_suffix_inside_tokens:
+            last_match_index = time_match[len(time_match)-1]
+            matches = self.config.location_matcher.find(text[0: last_match_index])
```
same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,153 @@
+        if len(time_match) != 0 and not is_all_suffix_inside_tokens:
+            last_match_index = time_match[len(time_match)-1]
+            matches = self.config.location_matcher.find(text[0: last_match_index])
+            location_matches = MatchingUtil.remove_sub_matches(matches.find(text))
```
same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,152 @@
+        if not self.config.location_time_suffix_regex:
+            return result
+
+        time_match = regex.match(self.config.location_time_suffix_regex, text)
```
re.match() only return a match object, if found. Based on C# version, I think here you have to obtain multiple Match objects.

Konigstiger replied:
You are right! the original implementation uses `.Matches`, that in C# returns `MatchCollection`. 
Instead, I think we can use this: 
`time_match = list(regex.finditer(self.config.location_time_suffix_regex, text))`
What do you think?

Maihav committed:
Finish to implement EnglishDateExtractorConfiguration constructor

Commit:
Applied Feedback

Changes
- Substituted `regex.match` with `regex.finditer`

Maihav commented on Python/libraries/recognizers-text/recognizers_text/utilities.py:
```diff
@@ -208,6 +208,23 @@ def apply_reverse(idx: int, string_chars, value: str):
+        # return normalized_query
+
+        # TODO: remove pass
+        pass
```
Please, don't commit commented code. Also, try to avoid the use of the `pass` sentence when you already started to implement the method. It's confusing.

Maihav committed:
Fix naming class

Maihav committed:
Implement TimeZoneExtractorConfiguration class on Spanish

Maihav committed:
Add newline at end of file

Maihav committed:
WIP: Update DateTimePeriod on English

Maihav committed:
Finish with FrenchTimeZoneExtractorConfiguration class

Maihav committed:
Add new line

Maihav committed:
WIP: Update DateTimePeriodExtractorConfiguration on English

Maihav committed:
Unify base_time_zone & base_timezone

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,152 @@
+            last_match_index = time_match[len(time_match)-1]
+            matches = regex.finditer(self.config.location_matcher, text[0: last_match_index])
+            location_matches = MatchingUtil.remove_sub_matches(regex.finditer(self.config.location_matcher, text[0: last_match_index]))
+
```
Why you are not using 'matches'? Here you have to pass 'matches' to _remove_sub_matches()_

Konigstiger replied:
Nice one. I did not saw it. Fixed.

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_time_zone.py:
```diff
@@ -0,0 +1,152 @@
+    def __init__(self, config: TimeZoneExtractorConfiguration):
+        self.config = config
+
+    def extract(self, source: str, reference: datetime = None) -> List[ExtractResult]:
```
You are missing an _extract()_ method in this class. Can you add it? Also, to implement it, use default values.

Maihav committed:
Merge branch 'python/timezone-migration/base-timezone-extractor-dev' into python/timezone-migration/base-timezone-extractor

Maihav committed:
Add missing extract method

Maihav committed:
Fix issue on timezone_extractor prop

Maihav committed:
Update Spanish DateTimePeriodExtractorConfig class

Commit:
Applied Feedback

Maihav committed:
Update FrenchDateTimePeriodExtractorConfiguration class

Commit:
Merge branch 'python/timezone-migration/base-timezone-extractor' into python/timezone-migration/base-timezone-extractor-dev

Commit:
Update FrenchTimePeriodExtractorConfiguration

Changes
Update FrenchTimePeriodExtractorConfiguration

Commit:
Applied Feedback

Changes for French
Feedback: applied

Commit:
PEP8
